### DISCUSSION TITLE
Why the class init is called twice ?

### DISCUSSION THREAD

**Discussion by yanel** [zauberzeug/nicegui#2844] [Q&A] [answered]
### Question

```python
from nicegui import ui

class TestClass:
	def __init__(self) -> None:
		self.name = 'Nicegui !'
		print('******************** INIT TESTCLASS *******************')


my_test_class = TestClass()

@ui.page('/')
def main_page():
	ui.label('Hello NiceGUI!')

ui.run()

```
### Output
******************** INIT TESTCLASS *******************
******************** INIT TESTCLASS *******************
NiceGUI ready to go on http://localhost:8080

**Comment by lemuriandezapada** [comment]
maybe you have the page open in two tabs

**Comment by rodja** [comment, marked as answer]
We answered the question in our FAQs: https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-code-executed-twice

**Reply by yanel** [reply]
Yes of course ! (shame on me)
Thanks.
